### PR TITLE
fix: Commands loading entrypoints

### PR DESCRIPTION
**Pull request by thetutlage**
This PR does a couple of changes

- It uses auto-detection to know if we are using `ts-node` to run the development server
- Next, it runs the `user` command in worker mode.

### CODE REVIEW

**Review by adrien2p — APPROVED** [review]

File: packages/medusa/src/commands/user.js [hunk 1]
```diff
@@ -20,6 +20,12 @@ export default async function ({
   track("CLI_USER", { with_id: !!id })
   const app = express()
   try {
+    /**
+     * Enabling worker mode to prevent discovering/loading
+     * starter kit resources
```
Comment (shahednasser) [inline packages/medusa/src/commands/user.js]: Maybe we should be clear here that it specifically doesn't register API routes, but not all resources

File: packages/medusa/src/loaders/config.ts [hunk 2]
```diff
@@ -102,8 +102,8 @@ export default (rootDirectory: string): ConfigModule => {
     modules: configModule.modules ?? {},
     featureFlags: configModule?.featureFlags ?? {},
     plugins: configModule?.plugins ?? [],
-    directories: configModule?.directories ?? {
-      srcDir: "dist",
+    directories: {
+      srcDir: process[Symbol.for("ts-node.register.instance")] ? "src" : "dist",
```
Comment (riqwan) [inline packages/medusa/src/loaders/config.ts]: is this really needed if its not used in a lot of places and not configurable?

Comment (adrien2p) [reply packages/medusa/src/loaders/config.ts]: We are curretly discussing it in slack :D

File: packages/medusa/src/commands/user.js (hunk 1 above)
Comment (thetutlage) [reply packages/medusa/src/commands/user.js]: Okay, is that the case. I thought it disables all the resources. Let me update the comment.

File: packages/medusa/src/loaders/config.ts (hunk 2 above)
Comment (thetutlage) [reply packages/medusa/src/loaders/config.ts]: Moved it to the place where it belongs. No longer in the config

File: packages/medusa/src/commands/user.js (hunk 1 above)
Comment (shahednasser) [reply packages/medusa/src/commands/user.js]: No just the API routes, it still loads modules and their resources

**Pull request merged**